geometric_kernels/kernels/geometric_kernels.py

```diff
@@ -15,16 +15,21 @@
 
 
 class MaternKarhunenLoeveKernel(BaseGeometricKernel):
-    r"""
-    This class approximates a kernel by the finite feature decomposition using
+    r"""This class approximates a kernel by the finite feature decomposition using
     its Laplace-Beltrami eigenfunctions and eigenvalues [1, 2].
 
-    .. math:: k(x, x') = \sum_{i=0}^{M-1} S(\sqrt\lambda_i) \phi_i(x) \phi_i(x'),
+    .. math:: k(x, x') = \sum_{i=0}^{M-1} S(\sqrt\lambda_i) \sum_{j=0}^{K_i} \phi_{ij}(x) \phi_{ij}(x'),
 
-    where :math:`\lambda_i` and :math:`\phi_i(\cdot)` are the eigenvalues and
+    where :math:`\lambda_i` and :math:`\phi_{ij}(\cdot)` are the eigenvalues and
     eigenfunctions of the Laplace-Beltrami operator and :math:`S(\cdot)` the
     spectrum of the stationary kernel. The eigenvalues and eigenfunctions belong
-    to the `SpaceWithEigenDecomposition` instance.
+    to the `DiscreteSpectrumSpace` instance.
+
+    We refer to sets of :math:`\{\phi_{ij}\}_{j=0}^{K_i}` as
+    "levels". In many practical spaces, like the sphere, we can employ
+    _addition theorems_ to efficiently compute the sums over levels
+    :math:`\sum_{j=0}^{K_i} \phi_{ij}(x) \phi_{ij}(x')`. A level may
+    consist of one eigenfunction.
```

Comment:
Maybe
-  "and eigenfunctions of the Laplace-Beltrami operator" -> " and eigenfunctions of the Laplace-Beltrami operator :math:`\Delta` such that :math:`\Delta \phi_{ij} = \lambda_i \phi_{ij}`".
- " We refer to sets of :math:`\{\phi_{ij}\}_{j=0}^{K_i}` as "levels". In many practical spaces, like the sphere, we can employ _addition theorems_ to efficiently compute the sums over levels :math:`\sum_{j=0}^{K_i} \phi_{ij}(x) \phi_{ij}(x')`. A level may consist of one eigenfunction."
  ->
   "We refer to the pairs :math:`(\lambda_i, G_i(\cdot, \cdot')` where :math:`G_i(\cdot, \cdot') = \sum_{j=0}^{K_i} \phi_{ij}(\cdot) \phi_{ij}(\cdot')` as "levels". For many spaces, like the sphere, we can employ _addition theorems_ to efficiently compute :math:`G_i(\cdot, \cdot')` without calculating the individual :math:`\phi_{ij}`. Note that although  :math:`\phi_{ij}` are guaranteed to correspond to the same eigenvalue :math:`\lambda_i`, the meaning of :math:`K_i` can vary, it does not have to be equal to the dimension of the whole eigenspace corresponding to :math:`\lambda_i`. For example, for `Graph` and `Mesh` it is always equal to 1 with eigenvalues having multiplicity greater than one giving rise to multiple levels with equal :math:`\lambda_i` but different :math:`G_i(\cdot, \cdot')`.".